CDP4Reporting/Behaviours/ReportDesignerBehaviour.cs

```diff
@@ -0,0 +1,202 @@
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Reporting.Behaviours
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Windows;
+    using DevExpress.Xpf.Ribbon;
+    using DevExpress.Mvvm.UI;
+    using DevExpress.Mvvm.UI.Interactivity;
+    using DevExpress.Xpf.Reports.UserDesigner;
+    using System.Diagnostics.CodeAnalysis;
+
+    /// <summary>
+    /// The purpose of the <see cref="ReportDesignerBehaviour"/> is to handle events from
+    /// the attached view of type <see cref="ReportDesigner"/>
+    /// </summary>
+    /// <remarks>
+    /// This behavior is meant to be attached to the ReportDesigner view from CDP4Reporting plugin
+    /// </remarks>
+    public class ReportDesignerBehaviour : Behavior<ReportDesigner>
+    {
+        /// <summary>
+        /// The ribbon merge category stored for cleanup.
+        /// </summary>
+        private RibbonPageCategoryBase mergeCategory;
+
+        /// <summary>
+        /// The ribbon merged categories stored for cleanup.
+        /// </summary>
+        private List<RibbonPageCategoryBase> mergedCategories;
+
+        /// <summary>
+        /// The main ribbon of the shell.
+        /// </summary>
+        private RibbonControl parentRibbon;
+
+        /// <summary>
+        /// The dependency property that allows setting the <see cref="IEventPublisher"/>
+        /// </summary>
+        public static readonly DependencyProperty RibbonMergeCategoryNameProperty = DependencyProperty.Register("RibbonMergeCategoryName", typeof(string), typeof(ReportDesignerBehaviour));
+
+        /// <summary>
+        /// Initializes static members of the <see cref="ReportDesignerBehaviour"/> class.
+        /// </summary>
+        static ReportDesignerBehaviour()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the name of the custom Ribbon Category to merge the reporting ribbon to.
+        /// </summary>
+        public string RibbonMergeCategoryName
+        {
+            get { return (string)this.GetValue(RibbonMergeCategoryNameProperty); }
+            set { this.SetValue(RibbonMergeCategoryNameProperty, value); }
+        }
+
+        /// <summary>
+        /// The on attached event handler
+        /// </summary>
+        [ExcludeFromCodeCoverage]
```

Comment:
Well, some things CAN be tested and there are indeed some tests for this class. 
Although it looks like it can be set on the class, setting it on individual methods is safer.